src/probe/probe.c

```diff
@@ -1491,6 +1491,37 @@ static int probe_set_config(struct processing_module *mod, uint32_t param_id,
 	}
 }
 
+static int probe_get_config(struct processing_module *mod,
+			    uint32_t config_id, uint32_t *data_offset_size,
+			    uint8_t *fragment, size_t fragment_size)
+{
+	struct sof_ipc_probe_info_params *info =
+		(struct sof_ipc_probe_info_params *)ASSUME_ALIGNED(fragment, 8);
+	struct probe_pdata *_probe = probe_get();
+	struct comp_dev *dev = mod->dev;
+	int i, j;
+
+	comp_dbg(dev, "probe_get_config() called config_id %u", config_id);
+	switch (config_id) {
+	case IPC4_PROBE_MODULE_PROBE_POINTS_ADD:
+		for (i = 0, j = 0; i < ARRAY_SIZE(_probe->probe_points); i++) {
+			if (_probe->probe_points[i].stream_tag == PROBE_POINT_INVALID)
+				continue;
+			if (offsetof(struct sof_ipc_probe_info_params, probe_point[j]) +
+			    sizeof(info->probe_point[0]) > fragment_size)
+				break;
+			info->probe_point[j++] = _probe->probe_points[i];
+		}
+		info->num_elems = j;
+		comp_info(dev, "probe_get_config() %u probe points sent", j);
```

Comment:
With Zephyr logging the function name is included in the log message automatically. So including it also explicitly just wastes memory. Are you getting it duplicated in your logs or is it somehow just my configuration? If you do get it twice, let's remove it. Would be great to clean up all Zephyr-only sources from this... In theory probes can be used with XTOS too IIUC? But hardly important enough to keep these names.